de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ArchiveIncomingProjectNegotiation.java

```diff
@@ -40,6 +41,8 @@
     private static final Logger LOG = Logger
         .getLogger(ArchiveIncomingProjectNegotiation.class);
 
+    private final ActivityQueuer activityQueuer = new ActivityQueuer();
```

Comment:
This changes many things at once, and I'm skeptic about them:

1. There is not one ActivityQueuer instance, but new queuers are created for each IPN.
2. Instead of better hiding the ActivityQueuer in the SarosSession (see deleted TODO below), the queuing becomes a shared responsibility of SarosSession and IPN.
3. There is an abstract super class, yet two implementations do very similar things with their respective queuers, which was before mostly part of the super class.

What are your reasons behind these changes?

Reply:
In general, because my changes do the opposite of the old TODOs and move queuing into the negotiation process.
In #156 I created a ResourceActivityQueuer which is called all the time from the incoming negotiation class.

regarding 3: I could have moved it to the abstract and than move it in the next patch and I know this patch would be cleaner this way, but it would have created new diff lines in the next patch, more work in rethinking and probably errors.
Thats the problem with stacked PRs..., I created a big prototype and just splitted what is somehow independent from each other.

regarding 1: I create new Queues for every IPN, because they could change when the next incoming negotiation uses a different type. I could make them static but didn't thought more about this and don't know if/when the class loader would drop the class and GC would take the queue.

regarding 2: For the resource queuing, I need to call queuing with SPaths as parameter. So instead of adding new methods to SarosSession, I decided to make it part of the only process which uses it and keep the declared methods amount of ISarosSession the same. This thereby makes it easier to add new functionality to the queuing implementation.

Reply:
Alright, I'm fine with that.

(Although I still find the way GitHub dispalys Pull Requests confusing, and all the more "stacked" Pull Requests. It's hard to get an overview.)